### ISSUE TITLE
Build Fails on Gatsby Cloud and locally with v3.4.0

### ISSUE THREAD

**Issue by ashhitch** [gatsbyjs/gatsby#31090]
## Description

When building on Gastby Cloud or locally with a site on v3.4.0 the build fails with `gatsby-plugin-gatsby-cloud` enabled.
 I don't have any serverless functions yet.
### Steps to reproduce

Update to gatsby 3.4.0 and push to Gatsby Cloud or run `gatsby build`.

### Expected result
Site should build without any serverless functions.

### Actual result

Build fails 
```
"gatsby-plugin-gatsby-cloud" threw an error while running the onPostBuild lifecycle:

ENOENT: no such file or directory, copyfile /usr/src/app/www/.cache/functions/manifest.json -> /usr/src/app/www/public/_functions.json
```

### Environment

This happens on Gatsby Cloud and locally
```
  System:
    OS: macOS 11.0.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.1 - ~/.nvm/versions/node/v14.15.1/bin/node
    Yarn: 1.21.1 - ~/.nvm/versions/node/v10.15.0/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.15.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 90.0.4430.85
    Firefox: 56.0.2
    Safari: 14.0.1
  npmPackages:
    gatsby: ^3.4.0 => 3.4.0
    gatsby-omni-font-loader: ^1.3.1 => 1.3.1
    gatsby-plugin-gatsby-cloud: ^2.4.0 => 2.4.0
    gatsby-plugin-google-tagmanager: ^3.3.0 => 3.4.0
    gatsby-plugin-graphql-config: ^0.13.0 => 0.13.0
    gatsby-plugin-image: ^1.4.0 => 1.4.0
    gatsby-plugin-layout: ^2.4.0 => 2.4.0
    gatsby-plugin-manifest: ^3.4.0 => 3.4.0
    gatsby-plugin-offline: ^4.4.0 => 4.4.0
    gatsby-plugin-react-helmet: ^4.4.0 => 4.4.0
    gatsby-plugin-sharp: ^3.4.0 => 3.4.0
    gatsby-plugin-styled-components: ^4.4.0 => 4.4.0
    gatsby-plugin-typescript: ^3.4.0 => 3.4.0
    gatsby-plugin-wpgraphql-seo: ^1.1.4 => 1.1.4
    gatsby-source-filesystem: ^3.4.0 => 3.4.0
    gatsby-source-wordpress: ^5.4.0 => 5.4.0
    gatsby-transformer-sharp: ^3.4.0 => 3.4.0
  npmGlobalPackages:
    gatsby-cli: 3.0.0

```

**Comment by LekoArts** [comment]
Do you have the flag enabled for functions or just `gatsby-plugin-gatsby-cloud` installed?

**Comment by ashhitch** [comment]
just gatsby-plugin-gatsby-cloud installed with no config

**Comment by bogdancss** [comment]
same issue here

**Comment by LekoArts** [comment]
The fix for that was published in `gatsby-plugin-gatsby-cloud@2.4.1`
